install/install.go

```diff
@@ -1316,7 +1317,18 @@ func (k *K8sInstaller) generateConfigMap() (*corev1.ConfigMap, error) {
 		m.Data["enable-endpoint-routes"] = "true"
 		m.Data["auto-create-cilium-node-resource"] = "true"
 		m.Data["enable-local-node-route"] = "false"
-		m.Data["masquerade"] = "false"
+
+		ersion := strings.TrimPrefix(k.params.Version, "v")
+		if v, err := semver.Parse(ersion); err != nil {
+			return nil, fmt.Errorf("unable to parse version %q: %w", k.params.Version, err)
```

Comment:
I think this will error out in case users provides their own custom version, e.g. a tag during developent. Could we just assume >=1.10.0 in this case and log a warning?

Reply:
i'm a bit torn about this one. warning might not be sufficient in this case, cilium might not function properly if we can't configure for a specific version 🤔

Reply:
Maybe we could just perform the check if the user didn't specifiy a custom agent and/or operator image, like in other places in the code. And assume >= 1.10 in case a custom agent/operator image was specified.

Otherwise I think something like `cilium install --version latest --agent-image quay.io/cilium/cilium-ci --operator-image quay.io/cilium/operator-azure-ci` would no longer work to test the lastest `master` build?